### PR TITLE
[DUOS-1814][risk=low] New API for updating select user fields

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1814

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by shaemarks — APPROVED** [review]
Code looks and functions great! Can confirm that I was able to update the fields listed in the ticket and I was not able to delete the researcher role, alter chair or member roles, or do any updates without an admin role

File: src/main/resources/assets/paths/userById.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 put:
+  deprecated: true
```
Comment (shaemarks) [inline src/main/resources/assets/paths/userById.yaml:2]: My Swagger page isn't showing this endpoint as deprecated for some reason

Comment (rushtong) [reply src/main/resources/assets/paths/userById.yaml:2]: Let me look into that - I don't think I checked for the deprecation in the ui.

Comment (rushtong) [reply src/main/resources/assets/paths/userById.yaml:2]: I'm seeing it as deprecated here:

Screen Shot 2022-07-11 at 5 24 49 PM

**Review by quazi-h — APPROVED** [review]
Changes here look good to me. I was able to use the new endpoint to update the specified fields. I tested by using an admin user to update the institutionId, suggestedInstitution, and eraCommonsId and confirned the changes applied to the user when querying /user/me.

**Pull request merged**